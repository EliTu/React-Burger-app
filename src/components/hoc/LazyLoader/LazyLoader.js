import React, { Suspense } from 'react';
import Spinner from '../../UI/Spinner/Spinner';

// A utility HOC for loading components lazily with React Router, implementing Suspense with a fallback for each component that is rendered through the Router
const LazyLoader = Component => {
	return props => (
		<Suspense fallback={<Spinner />}>
			<Component {...props} />
		</Suspense>
	);
};

export default LazyLoader;
