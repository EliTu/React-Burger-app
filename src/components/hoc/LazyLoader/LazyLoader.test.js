import React from 'react';
import LazyLoader from './LazyLoader';
import { shallow } from 'enzyme';

describe('<LazyLoader>', () => {
	const testComponent = props => <div {...props}>testComponent</div>;
	const lazy1 = new Promise((resolve, reject) => {
		setTimeout(() => {
			resolve(testComponent);
		}, 1000);
	});

	it('should render a component without errors', async () => {
		const loaded1 = await lazy1.then(loaded => loaded);

		let wrapper = shallow(<LazyLoader component={loaded1} />);
		expect(wrapper.html()).toMatchSnapshot();
		expect(wrapper.length).toBe(1);
	});
});
