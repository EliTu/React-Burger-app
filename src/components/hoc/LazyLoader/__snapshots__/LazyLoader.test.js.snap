// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<LazyLoader> should render a component without errors 1`] = `""`;
